_pages/model-deployment/xgboost.md

```diff
@@ -164,3 +173,129 @@ If you don't recall from the <a href="{{site.baseurl}}/algorithm-development/alg
 If you want to have a better idea of what a finished XGBoost algorithm looks like loading a XGBoost model, check out: <a href="https://algorithmia.com/algorithms/demo/scikitlearnxgboostdemo">scikitlearnxgboostdemo</a>.
 
 That's it for hosting your XGBoost model on Algorithmia!
+
+
+## Deploying From Within Jupyter Notebook
+This section demonstrates how you can create an algorithm on Algorithmia, push your algorithm script, your dependency file and your saved model file to Algorithmia all programmatically from within a Jupyter notebook.
+
+
+If you'd like to follow along this tutorial and reproduce the steps, you can clone the example notebook under the repository at [XGBoost Jupyter Notebook Demo](https://github.com/algorithmiaio/model-deployment/tree/master/xgboost_notebook_to_algorithmia). This example contains:
+- The training data 
+- Python utility functions to help with our programmatic interactions with Algorithmia
+- The runnable Jupyter notebook
+
+Before you get started, you'll also want to make sure that you have all the imported Python packages on your development environment.
+
+
+### Creating an Algorithm
+
+For these operations, we will use the utility functions defined on our imported [Algorithmia utility script](https://github.com/algorithmiaio/model-deployment/blob/master/xgboost_notebook_to_algorithmia/algorithmia_utils.py) This script encapsulates the related calls to Algorithmia, through its [Python API](https://algorithmia.com/developers/clients/python).
+
+
+```python
+import Algorithmia
+import algorithmia_utils
+```
+
+```python
+api_key = "YOUR_API_KEY"
```

Comment:
We have macros to fill in the user's actual name and API keys here. There are examples elsewhere in the dev-center.